ERCS/erc-7715.md

```diff
@@ -0,0 +1,466 @@
+type NativeTokenLimitPermission = {
+  type: 'native-token-limit',
+  data: {
+    amount: "0x...",  // wei in hex
+  }
+}
+
+// ERC20 spending limit
+type ERC20LimitPermission = {
+  type: 'erc20-limit',
+  data: {
+    erc20Address: "0x..."  // ERC20 token address
+    amount: "0x...",  // wei in hex
+  }
+}
+
+// Gas spending limit
+type GasLimitPermission = {
+  type: 'gas-limit',
+  data: {
+    amount: "0x...",  // wei in hex
+  }
+}
+
+// Allowlist of contract calls
+type ContractCallPermission = {
+  type: 'contract-call',
+  data: {
+    // TODO
+  }
+}
+
+// The number of calls the session can make during each interval
+type RateLimitPermission = {
+  type: 'rate-limit',
+  data: {
+    count: number,  // the number of times during each interval
+    interval: number,  // in seconds
+  }
+}
+```
+
+### Wallet-managed Sessions
+
+If `signer` is not specified in the request, the wallet manages the session.  If the wallet approves the request, it MUST accept [ERC-5792](./eip-5792.md)’s `wallet_sendCalls` with the `permissions` capability, which MAY include the session with a `permissionsContext`.  For example:
+
+```tsx
+[
+  {
+    "version": "1.0",
+    "chainId": "0x01",
+    "from": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+    "calls": [
+      {
+        "to": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+        "value": "0x9184e72a",
+        "data": "0xd46e8dd67c5d32be8d46e8dd67c5d32be8058bb8eb970870f072445675058bb8eb970870f072445675"
+      },
+      {
+        "to": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+        "value": "0x182183",
+        "data": "0xfbadbaf01"
+      }
+    ],
+    "capabilities": {
+      "permissions": {
+        "context": "<permissionsContext>"
+      }
+    }
+  }
+]
+```
+
+Upon receiving this request, the wallet MUST send the calls in accordance with the requested permissions.  The wallet SHOULD NOT ask the user for further transaction confirmations.
+
+### Capabilities
+
+If the wallet supports [ERC-5792](./eip-5792.md), wallet SHOULD respond on **`wallet_getCapabilities`** request using the `permissions` key.
+
+The wallet SHOULD include `signerTypes` (`string[]`) and `permissionTypes` (`string[]`) in the response, to specify the signer types and permission types it supports.
+Example:
+
+```json
+{
+  "0x123": {
+    "permissions": {
+      "supported": true,
+      "signerTypes": ["keys", "account"],
+      "permissionTypes": ["erc20-limit", "contract-call"]
+    },
+  },
+}
+```
+
+If the wallet is using CAIP-25 authorization, wallet SHOULD include `permissions` key in the CAIP-25 `sessionProperties` object.  Additional keys to include are  `permissionTypes` with the comma separated list of supported permission types and `signerTypes` with the comma separated list of supported signer types.
+
+Example:
+
+```json
+{
+  //...
+  "sessionProperties": {
+    "permissions": "true",
+    "signerTypes": "keys,account",
+    "permissionTypes": "erc20-limit,contract-call"
```

Comment:
Do these need to be serialized like this? Why not an array?

Reply:
CAIP-25 only supports `<string>:<string> `type object

Reply:
CAIP-25 is the most efficient way to discover capabilities since it eliminates additional network call.